### PR TITLE
Feature/proof of address UK CX-2254

### PR DESCRIPTION
**Pull request by elhesuu**
# Problem
Selecting a document, user should see a capture screen mentioning the PoA type. Upon confirmation, the file will uploaded to the API with UK as the country and the corresponding PoA type as the document type `utility_bill`, `council_tax`, `benefits_letter`, `bank_statement`, `credit_card_statement`.

# Solution
- Add PoA documents to document selector
- Update ID documents selector to match new design

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [n/a] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [n/a] Have new automated tests been implemented?
- [n/a] Have new manual tests been written down?
- [x] Have tests passed locally?
- [x] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by seewah** [comment]
btw @elhesuu, also i mentioned on slack, but it should be `bank_building_society_statement` instead of `bank_statement`

Also, i checked with the doc team, and they say it should be `benefit_letters` instead of letter (singular). @rfreitas what do you think? I do not agree with this inconsistency, and I don't want us to introduce something that may break later, but at the same time, my concern seems to fall on deaf ears...

**Comment by seewah** [comment]
terribly sorry @elhesuu i checked with @danielspagnolo, and the overall left right margin should be 16px instead of 8px (so that it is consistent with all the other screens). The marvelapp screens should have shown 8px instead of 16px

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
@seewah so your concern is that it's the only document type which is plural rather than singular?

**Comment by seewah** [comment]
sorry @elhesuu i missed another UI glitch yesterday. The document list should always be anchored top-aligned and not middle-aligned. The list should sit 16px underneath "Select the type of document you would like to upload" in the mobile view, and 24px in the desktop view, as per marvel screens. Thanks!

**Comment by seewah** [comment]
@rfreitas yes, that is my concern

a minor point is i think it is benefits rather than benefit...

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/ProofOfAddress/PoAIntro/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import classNames from 'classnames'
+import theme from '../../Theme/style.css'
+import style from './style.css'
+import Title from '../../Title'
+import {trackComponent} from '../../../Tracker'
+import {preventDefaultOnClick} from '../../utils'
+import {parseI18nWithXmlTags} from '../../../locales'
+
+const PoAIntro = ({i18n, nextStep}) => {
+  return (
```
Comment (rfreitas) [inline src/components/ProofOfAddress/PoAIntro/index.js]: Minor: redundant `return`

File: src/components/DocumentSelector/index.js [hunk 2]
```diff
@@ -30,32 +33,104 @@ class DocumentSelector extends Component {
...
-        {documentOptions.map((op) => this.renderOption(op))}
+      <div className={classNames(style.wrapper, className)}>
+        {documentOptions.map(this.renderOption)}
       </div>
     )
   }
 }
 
-export default DocumentSelector;
+const snakeCase = str => str.replace(/_/g, '-')
```
Comment (rfreitas) [inline src/components/DocumentSelector/index.js]: minor: snakeToDashCase ?

File: src/components/DocumentSelector/index.js [hunk 3]
```diff
@@ -1,26 +1,29 @@
 import { h, Component } from 'preact'
 import style from './style.css'
+import theme from '../Theme/style.css'
+import classNames from 'classnames'
+
+type DocumentTypeOption = {
+  eStatementAccepted?: string,
```
Comment (rfreitas) [inline src/components/DocumentSelector/index.js]: isn't this a boolean?

Comment (elhesuu) [reply src/components/DocumentSelector/index.js]: Yes, thanks @rfreitas

File: src/components/DocumentSelector/index.js (hunk 2 above)
Comment (elhesuu) [reply src/components/DocumentSelector/index.js]: @rfreitas, I named it `snakeCase`, but really I wanted to name it `kebabCase`. What do you think about `snakeToKebabCase`?

Comment (rfreitas) [reply src/components/DocumentSelector/index.js]: I think snakeToKebab is probably the best function name I've come across.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

File: src/locales/es.json [hunk 4]
```diff
@@ -6,8 +6,13 @@
...
   "document_selector": {
-    "title": "Verifique su identidad",
-    "hint": "Seleccione el tipo de documento que desea subir"
+    "identity": {
+      "title": "Verifique su identidad",
+      "hint": "Seleccione el tipo de documento que desea subir",
+      "passport_hint": "Página del pasaporte con su foto",
+      "driving_licence_hint": "Frente y reverso",
+      "national_identity_card_hint": "Frente y reverso"
+    }
```
Comment (stefaniacardenas) [inline src/locales/es.json]: I can see a few strings missing, are they still to be translated?

Comment (elhesuu) [reply src/locales/es.json]: Translation is not needed for PoA documents for now, so I only added translations for those strings that are needed for ID documents

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by elhesuu** [comment]
@rfreitas, just checked CSS size diff and with these changes it's actually a few bytes smaller.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by seewah** [comment]
as discussed on slack please make sure that glare detection is on for any of the PoA types. If possible, i would suggest we display the "make sure ... is readable" message for all confirmation screens too, regardless of id or PoA.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by seewah** [comment]
another issue i noticed. PoA should not undergo "id document detection" either. Basically PoA should not undergo any image validation whatsoever

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-392-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://392-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**